app/controllers/student_organizations_controller.ts

```diff
@@ -0,0 +1,39 @@
+import type { HttpContext } from "@adonisjs/core/http";
+
+import StudentOrganization from "#models/student_organization";
+import { paginationValidator } from "#validators/pagination";
+import { showValidator } from "#validators/show";
+
+export default class StudentOrganizationsController {
+  /**
+   * Display a list of resource
+   */
+  async index({ request }: HttpContext) {
+    const { page, limit } = await request.validateUsing(paginationValidator);
+    return StudentOrganization.query()
+      .preload("links")
+      .preload("tags")
+      .withScopes((scopes) => {
+        scopes.handleSearchQuery(request.qs(), "source", "organizationType");
+        scopes.handleSortQuery(request.input("sort"));
```

Comment:
```suggestion
      .withScopes((scopes) => {
        scopes.handleSearchQuery(request.qs(), "source", "organizationType");
        scopes.handleSortQuery(request.input("sort"));
        scopes.preloadRelations(request.only(["links", "tags"]));
      })
```
If it's not always mandatory to fetch `links` and `tags` with `departaments`
You can add something like this, to leave "preloading" optional for the client to decide.

Reply:
but you need to implement this scope same like other, in the model first